packages/blockchain-extension/extension/commands/importNodesToEnvironmentCommand.ts

```diff
@@ -117,13 +117,51 @@ export async function importNodesToEnvironment(environmentRegistryEntry: FabricE
             if (!apiKey) {
                 return;
             }
+            const apiSecret: string = await UserInputUtil.showInputBox('Enter the api secret of the ops tools you want to connect to');
+            if (!apiSecret) {
+                return;
+            }
 
+            const api: string = url.replace(/\/$/, '') + GET_ALL_COMPONENTS;
             let response: any;
+            const requestOptions: any = {
+                headers: { 'Content-Type': 'application/json' },
+                auth: { username: apiKey, password: apiSecret }
+            };
             try {
-                const api: string = url.replace(/\/$/, '') + GET_ALL_COMPONENTS;
-                response = await Axios.get(api, {
-                    headers: { Authorization: `Bearer ${apiKey}` }
-                });
+                try {
+                    response = await Axios.get(api, requestOptions);
+                } catch (error) {
+                    // This needs to be fixed - exactly what codes can we get that will require this behaviour?
+                    if (error.code === 'DEPTH_ZERO_SELF_SIGNED_CERT') {
+                        const certificateUsage: string = await UserInputUtil.showQuickPick('Unable to perform certificate verification. Please choose how to proceed', [UserInputUtil.ADD_CA_CERT_CHAIN, UserInputUtil.CONNECT_NO_CA_CERT_CHAIN]) as string;
+                        if (!certificateUsage) {
+                            return;
+                        } else if (certificateUsage === UserInputUtil.ADD_CA_CERT_CHAIN) {
+                            const quickPickItems: string[] = [UserInputUtil.BROWSE_LABEL];
+                            const browseOptions: vscode.OpenDialogOptions = {
+                                canSelectFiles: true,
+                                canSelectFolders: false,
+                                canSelectMany: false,
+                                openLabel: 'Select',
+                                filters: {
+                                    Certificates: ['pem']
+                                }
+                            };
+                            const certificatePath: vscode.Uri = await UserInputUtil.browse('Enter a file path to the collection configuration', quickPickItems, browseOptions, true) as vscode.Uri;
```

Comment:
Perhaps we should change the message shown to the user (as they aren't entering a file path, but choosing a file instead?) to something like:
`Select the certificate (pem) you wish to add`

Reply:
Well spotted :)